datalad_next/runners/tests/test_batch.py

```diff
@@ -0,0 +1,30 @@
+from ..batch import (
+    annexjson_batchcommand,
+    stdout_batchcommand,
+)
+
+
+def test_batch_simple(existing_dataset):
+    with stdout_batchcommand(
+        ['git', 'annex', 'examinekey',
+         # the \n in the format is needed to produce an output that hits
+         # the output queue after each input line
+         '--format', '${bytesize}\n',
+         '--batch'],
+        cwd=existing_dataset.pathobj,
+    ) as bp:
+        res = bp.proc(b'MD5E-s21032--2f4e22eb05d58c21663794876dc701aa\n')
+        assert res.rstrip(b'\r\n') == b'21032'
+        res = bp.proc(b'MD5E-s999--2f4e22eb05d58c21663794876dc701aa\n')
+        assert res.rstrip(b'\r\n') == b'999'
+
+    with annexjson_batchcommand(
+        ['git', 'annex', 'examinekey', '--json',
+         '--batch'],
+        cwd=existing_dataset.pathobj,
+    ) as bp:
+        res = bp.proc(b'MD5E-s21032--2f4e22eb05d58c21663794876dc701aa\n')
+        # TODO the outcome should be the "result" of the protocol, which
+        # would be a decoded JSON data structure, not just the string
+        # read from the process stdout
+        assert res.rstrip(b'\r\n') == b'{"backend":"MD5E","bytesize":"21032","error-messages":[],"hashdirlower":"20f/01d/","hashdirmixed":"00/7F/","humansize":"21.03 kB","key":"MD5E-s21032--2f4e22eb05d58c21663794876dc701aa","keyname":"2f4e22eb05d58c21663794876dc701aa","mtime":"unknown","objectpath":".git/annex/objects/00/7F/MD5E-s21032--2f4e22eb05d58c21663794876dc701aa/MD5E-s21032--2f4e22eb05d58c21663794876dc701aa","objectpointer":"/annex/objects/MD5E-s21032--2f4e22eb05d58c21663794876dc701aa\\n"}'
```

Comment:
@christian-monch what would need to change in order to get the actual protocol result here?